tempodb/wal/wal.go

```diff
@@ -97,13 +82,19 @@ func New(c *Config) (*WAL, error) {
 }
 
 // RescanBlocks returns a slice of append blocks from the wal folder
-func (w *WAL) RescanBlocks(fn RangeFunc, additionalStartSlack time.Duration, log log.Logger) ([]*AppendBlock, error) {
+func (w *WAL) RescanBlocks(additionalStartSlack time.Duration, log log.Logger) ([]common.WALBlock, error) {
 	files, err := os.ReadDir(w.c.Filepath)
 	if err != nil {
 		return nil, err
 	}
 
-	blocks := make([]*AppendBlock, 0, len(files))
+	// todo: rescan blocks will need to detect if this is a vParquet or v2 wal file and choose the appropriate encoding
```

Comment:
I think ultimately we will have to decide whether the wal files are owned by tempodb/wal or the encoding and that determines what we do here.  This is good for now as we iterate.

Reply:
I like having a wal object that is responsible for creating and managing the wal blocks. Agree we can revisit in the future if it gets too thin.